### PR TITLE
Update how Eden size is reported at gc-start and gc-end

### PR DESCRIPTION
**Pull request by bragaigor**
Fix how Eden size is reported at `gc-start` and `gc-end` in balanced GC policy. Make it consistent with reporting
MM_HeapRegionManagerVLHGC::getHeapMemorySnapshot()

Reorder how/when taxation threshold and regions' age are updated. Instead of updating region's age at the very end of a GC increment, we now update it at the end of GC increment but before we report `gc-end`. With the update of `exportStats` on how Eden is reported, we now have total Eden size being reported as `getCurrentEdenSizeInBytes()` always while free eden size in bytes contains the correct value, where at `gc-start` `_edenFreeHeapSize` is close to zero and at `gc-end` it's equal to Eden heap size.

Signed-off-by: Igor Braga <higorb1@gmail.com>

### CODE REVIEW

**Comment by bragaigor** [comment]
cc: @amicic @LinHu2016 @dmitripivkine

**Comment by amicic** [comment]
While this accomplishes that we want (reports 100% free in Verbose GC at the end of PGC ), this might still not be good enough if used at GMP increments points (we can't reliably distinguish these points from PGC end, just based on timestamps that are not PGC specific, but increment generic).

Additionally, let's try to see if this can work without special treating PGC end althogether. The reason we need that special treatment is because ages are incremented after PGC end is reported, what leads to survivor and marked regions being incorrectly treated as age 0 regions, when export stats is invoked at PGC end. 

Let's try to increment ages before PGC end and see if it will help (or cause more troubles than we have with this special treatment).

File: runtime/gc_vlhgc/IncrementalGenerationalGC.cpp [hunk 1]
```diff
@@ -2101,6 +2101,9 @@ MM_IncrementalGenerationalGC::exportStats(MM_EnvironmentVLHGC *env, MM_Collectio
 		/* numaNodes is just used as indication to verbose GC that stats we collected are valid and indeed should be reported */
 		stats->_numaNodes = _extensions->_numaManager.getAffinityLeaderCount();
 		UDATA regionSize = _regionManager->getRegionSize();
+		UDATA allocateEdenTotal = 0;
+		stats->_edenHeapSize = getCurrentEdenSizeInBytes(env);
+		bool isGCend = stats->_startTime > stats->_endTime;
```
Comment (bragaigor) [inline runtime/gc_vlhgc/IncrementalGenerationalGC.cpp:2106]: Remove

File: runtime/gc_vlhgc/IncrementalGenerationalGC.cpp [hunk 2]
```diff
@@ -2111,23 +2113,20 @@ MM_IncrementalGenerationalGC::exportStats(MM_EnvironmentVLHGC *env, MM_Collectio
...
+						allocateEdenTotal += regionSize;
 					} else {
 						usedMemory = regionSize - memoryPool->getFreeMemoryAndDarkMatterBytes();
 					}
 				} else {
 					Assert_MM_true(region->isArrayletLeaf());
 					usedMemory = regionSize;
-					/* for Eden arraylet leaf Allocation Age must be smaller then amount allocated since last PGC */
-					if (region->getAllocationAge() <= _allocatedSinceLastPGC) {
-						stats->_edenHeapSize += regionSize;
```
Comment (amicic) [inline runtime/gc_vlhgc/IncrementalGenerationalGC.cpp:2129]: we still need to update allocateEdenTotal, even for arraylet (age 0) region

File: runtime/gc_vlhgc/IncrementalGenerationalGC.cpp [hunk 3]
```diff
@@ -2111,23 +2113,20 @@ MM_IncrementalGenerationalGC::exportStats(MM_EnvironmentVLHGC *env, MM_Collectio
 				if (region->containsObjects()) {
 					MM_MemoryPoolBumpPointer *memoryPool = (MM_MemoryPoolBumpPointer *)region->getMemoryPool();
 					Assert_MM_true(NULL != memoryPool);
-					/* for Eden region containing objects Allocation Age must be smaller then amount allocated since last PGC */
-					if (region->getAllocationAge() <= _allocatedSinceLastPGC) {
-						stats->_edenHeapSize += regionSize;
+					/* Eden region containing objects, Allocation Age must be smaller then amount allocated since last PGC,
+					 * more accurately, its logical age must be equal to zero */
+					if (0 == region->getLogicalAge()) {
```
Comment (amicic) [inline runtime/gc_vlhgc/IncrementalGenerationalGC.cpp:2118]: I prefer to use the old age condition (it's more generic), which might work better after reordering in https://github.com/eclipse/openj9/pull/11378

**Review by amicic — APPROVED** [review]

**Comment by amicic** [comment]
Jenkins compile all jdk8

**Comment by amicic** [comment]
Jenkins sanity test Win Jdk11

**Pull request merged**